src/ofdm.c

```diff
@@ -749,6 +749,48 @@ static float est_freq_offset(struct OFDM *ofdm, complex float *rx, int timing_es
     return foff_est;
 }
 
+
+static float est_freq_offset_pilot_corr(struct OFDM *ofdm, complex float *rx, int timing_est) {
+    complex float corr_st[ofdm_m + ofdm_ncp], corr_en[ofdm_m + ofdm_ncp];
+    int SFrame = ofdm_samplesperframe;
+
+    // "mix" down (correlate) the pilot sequences from frame with 0 Hz offset pilot samples
+    for (int i = 0; i < (ofdm_m + ofdm_ncp); i++) {
+        complex float csam = conjf(ofdm->pilot_samples[i]);
+        corr_st[i] = rx[timing_est + i         ] * csam;
+        corr_en[i] = rx[timing_est + i + SFrame] * csam;
+    }
```

Comment:
@srsampson we could also combine this step with the correlation below, removing the need for storage of corr_st[] and corr_en[]

Reply:
Yes, you're going to get two multiplies inside the loop either way. I've made the change, so see if it works.

Reply:
Nice work Steve, ctests pass so looks good.

I've implemented a high/low bandwidth phase estimation scheme, an extension on your high_doppler support.  Lets us acquire when the freq offset est is a bit off.  We "switch gears" using the state machine to retain the low SNR performance of the low bandwidth phase est.

Only problem is I broke your manual phase est bandwidth setting (formally --high-doppler option in ofdm_demod).  It would be nice to set the bandwidth high/low/auto from the command line.  Could you work on that pls?  I've highlighted the code that needs some work.

Reply:
> It would be nice to set the bandwidth high/low/auto from the command line. Could you work on that pls? I've highlighted the code that needs some work.

I'll take a look and play with it.